src/dnn/MIOpen.jl

```diff
@@ -0,0 +1,77 @@
+module MIOpen
+
+import AMDGPU
+import AMDGPU.Runtime.Mem
+
+using AMDGPU: ROCArray, ROCDevice
+using CEnum
+using GPUArrays
+using MIOpen_jll
+
+const libMIOpen_path = MIOpen_jll.libMIOpen_path
+
+include("low_level.jl")
+
+const HANDLE = Ref{miopenHandle_t}(C_NULL)
+
+function get_status_string(status)
+    if     status == miopenStatusSuccess return "Success"
+    elseif status == miopenStatusNotInitialized return "Not initialized"
+    elseif status == miopenStatusInvalidValue return "Invalid value"
+    elseif status == miopenStatusBadParm return "Bad parameter"
+    elseif status == miopenStatusAllocFailed return "Allocation failed"
+    elseif status == miopenStatusInternalError return "Internal error"
+    elseif status == miopenStatusNotImplemented return "Not implemented"
+    elseif status == miopenStatusUnknownError return "Unknown error"
+    elseif status == miopenStatusUnsupportedOp return "Unsupported operation"
+    elseif status == miopenStatusGpuOperationsSkipped return "GPU operations skipped" end
+    error("Invalid status code: $status")
+end
+
+function check(status)
+    if status != miopenStatusSuccess
+        status_string = get_status_string(status)
+        error("MIOpen return status [code=$status]: $status_string")
```

Comment:
Maybe we can make an `MIOpenException` object (which contains the numeric status code) and throw that instead, so that it's easier for libraries to catch MIOpen errors? Then this string can be part of the `Base.showerror` implementation.